src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1487 @@
+
+            Error array will be assigned to ``'err'`` key of the output model.
+
+            .. note::
+                At this time, output error array is not equivalent to
+                error propagation results.
+
+        """
+        # always required:
+        attributes = {
+            "data",
+            "wcs",
+            "wht",
+        }
+
+        if enable_ctx:
+            attributes.add("con")
+        if compute_err:
+            attributes.add("err")
+        if enable_var:
+            attributes.update(
+                ["var_rnoise", "var_poisson", "var_flat"]
+            )
+            # TODO: if we want to support adding more data to
+            # existing output models, we need to also store weights
+            # for variance arrays:
+            # var_rnoise_weight
+            # var_flat_weight
+            # var_poisson_weight
+        if accumulate:
+            if enable_ctx:
+                attributes.add("n_coadds")
+
+            # additional attributes required for input parameter 'output_model'
+            # when data and weight arrays are not None:
+            attributes.update(
+                {
+                    "pixfrac",
+                    "kernel",
+                    "fillval",
+                    "weight_type",
+                    "pointings",
+                    "exposure_time",
+                    "measurement_time",
+                    "start_time",
+                    "end_time",
+                    "duration",
+                }
+            )
+
+        return attributes
+
+    def check_output_wcs(self, output_wcs, estimate_output_shape=True):
+        """
+        Check that provided WCS has expected properties and that its
+        ``array_shape`` property is defined.
+
+        Parameters
+        ----------
+        output_wcs : WCS object
+            A WCS object corresponding to the output (resampled) image.
+
+        estimate_output_shape : bool, optional
+            Indicates whether to *estimate* pixel scale of the ``output_wcs``
+            from
+
+        """
+        naxes = output_wcs.output_frame.naxes
+        if naxes != 2:
+            raise UnsupportedWCSError(
+                "Output WCS needs 2 coordinate axes but the "
+                f"supplied WCS has {naxes} axes."
+            )
+
+        # make sure array_shape and pixel_shape are set:
+        if output_wcs.array_shape is None and estimate_output_shape:
+            # if wcs_pars and "output_shape" in wcs_pars:
+            #     output_wcs.array_shape = wcs_pars["output_shape"]
+            # else:
+            if output_wcs.bounding_box:
+                halfpix = 0.5 + sys.float_info.epsilon
+                output_wcs.array_shape = (
+                    int(output_wcs.bounding_box[1][1] + halfpix),
+                    int(output_wcs.bounding_box[0][1] + halfpix),
+                )
+            else:
+                # TODO: In principle, we could compute footprints of all
```

Comment:
Can this TODO (or any other ones in this code) be removed and if desired, turned into a GitHub issue or JIRA ticket so we can track work still needed?

Reply:
It could, but does it make sense to open a ticket to modify code that does not yet exist in the repository.

Reply:
It's not necessarily my decision but I would say yes, because this is definitely going to get added at some point and it's easier to track issues than code comments